pkg/apis/azuredisk/v1alpha1/types.go

```diff
@@ -163,6 +175,13 @@ type AzVolumeAttachmentList struct {
 	Items []AzVolumeAttachment `json:"items"`
 }
 
+type AzError struct {
```

Comment:
@sunpa93 Did you take #57 in part or whole for this change? I would prefer we merge the other PR first. @nearora-msft when do you believe you can get the error handling changes in?

Reply:
Testing out my changes. Targeting to merge by eod.

Reply:
It merged #57 in part. In order to handle dangling attach error in `controller_server`, AzError PR was necessary. I will rebase once #57 is merged and resolve any conflict raised from it.